src/main/java/core/basesyntax/service/operation/BalanceOperationHandlerImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperationHandlerImpl implements OperationHandler {
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        if (transaction == null) {
+            throw new RuntimeException("Transaction is invalid");
+        }
+        String fruitName = transaction.getFruit();
+        int quantityFruit = transaction.getQuantity();
+        if (quantityFruit < 0) {
+            throw new RuntimeException("Quantity must be greater or equals 0!");
+        }

```

Comment:
negative ballance is incorrect for any transaction object
adding the same code check to each OperationHandler is not SOLID- and DRY-friendly
Simply do it in the parser